src/lib/utils/meta_tables/meta_settings_table.cpp

```diff
@@ -0,0 +1,41 @@
+#include "meta_settings_table.hpp"
+
+#include "hyrise.hpp"
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+MetaSettingsTable::MetaSettingsTable()
+    : AbstractMetaTable(TableColumnDefinitions{{"setting_name", DataType::String, false},
+                                               {"value", DataType::String, false},
+                                               {"description", DataType::String, false}}) {}
+
+const std::string& MetaSettingsTable::name() const {
+  static const auto name = std::string{"settings"};
+  return name;
+}
+
+bool MetaSettingsTable::can_update() const { return true; }
+
+std::shared_ptr<Table> MetaSettingsTable::_on_generate() const {
+  auto output_table = std::make_shared<Table>(_column_definitions, TableType::Data, std::nullopt, UseMvcc::Yes);
+  const auto setting_names = Hyrise::get().settings_manager.all_settings();
```

Comment:
I would have expected `all_settings` to return a map (or similar). Maybe `setting_names`?